### PR TITLE
Cleanup building mobile extensions section

### PR DESCRIPTION
**Pull request by praveek**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by calebk1 — APPROVED** [review]
LGTM!

Just want to double check - do we link to the current buildling-mobile-extensions anywhere else? If we do, we should get rid of those links as well.

File: src/pages/documentation/building-mobile-extensions/interacting-with-rules.md [hunk 1]
```diff
@@ -1,95 +0,0 @@
-# Interacting with rules
-
-Rules are defined in the Data Collection UI interface and can include data elements, events, conditions, and actions provided by extensions.
-
-## Publishing rules support for tags
-
-Your extension should publish the rules support that it provides to tag libraries. You can provide this support through the `extension.json` file in your extensions base directory.
-
-### Publishing supported data elements
```
Comment (emdobrin) [inline src/pages/documentation/building-mobile-extensions/interacting-with-rules.md:9]: do we have some rule examples for extensions in the github docs?

File: src/pages/documentation/building-mobile-extensions.md [hunk 2]
```diff
@@ -8,6 +8,25 @@ To support customer-specific features, and allow for the greatest flexibility, A
 
 Extensions allow customers to extend the Experience Platform SDKs with their own code. This includes listening for and dispatching events, reading the shared state of any registered extension, and sharing the state of the current extension. The application can use the extension to monitor for information that Adobe does not expose by default. It can also use the extension to modify Experience Platform SDK internal operations. For example, the application can use the extension to add additional data to messages that are sent or by sending data to other systems.
 
+## Prerequisites to building an extension
+
+Before you build an extension, complete the following tasks:
+
+### Ensure that you are using the Adobe Experience Platform SDKs.
```
Comment (emdobrin) [inline src/pages/documentation/building-mobile-extensions.md:15]: Nit: can you remove the periods from these two headers

File: src/pages/documentation/building-mobile-extensions.md [hunk 3]
```diff
@@ -22,45 +41,14 @@ The following naming rules apply for extensions:
...
-* When using a deprecated shared state name.
-* When registration is attempted during extension shutdown.
-* When an event is being dispatched while the extension is being shut down.
-* When a callback from the Experience Platform SDKs to the external code throws an exception.
-* When an internal error occurs, or an unexpected exception is thrown.
-
-<InlineAlert variant="warning" slots="text"/>
 
-In iOS, asynchronous errors are handled by using the `unexpectedError` method that is defined in the `ACPExtension` class.
+For help on building your own extension, please see:
```
Comment (emdobrin) [inline src/pages/documentation/building-mobile-extensions.md:51]: let's add a H2 header before this line: Get started / Build extensions

File: src/pages/documentation/building-mobile-extensions/interacting-with-rules.md (hunk 1 above)
Comment (praveek) [reply src/pages/documentation/building-mobile-extensions/interacting-with-rules.md:9]: Not currently but most of this is covered in the https://experienceleague.adobe.com/docs/experience-platform/tags/extension-dev/overview.html?lang=en

File: src/pages/documentation/building-mobile-extensions.md [hunk 4]
```diff
@@ -8,6 +8,25 @@ To support customer-specific features, and allow for the greatest flexibility, A
...
+  To determine your goals, think about the following questions:
+
+  * Do you need access to data that is not already exposed via the Adobe Experience Platform SDKs?    
+  * Do you need to be notified when messages will be sent, or data is being collected by the Adobe Experience Platform SDKs?   
+  * Do you need to add data to or modify data for outgoing messages?    
+  * Do you need to expose data to other extensions or to rules processing?
+
+  If your answer to any of these questions is **yes**, build the extension.
+
 ## Namespace conventions
```
Comment (emdobrin) [inline src/pages/documentation/building-mobile-extensions.md:30]: nit: can we use h3 here as a prerequisite for building extensions and rename to "Review the namespace conventions"

**Review by emdobrin — APPROVED** [review]
One small comment, looks good otherwise

File: src/pages/documentation/building-mobile-extensions/interacting-with-rules.md (hunk 1 above)
Comment (emdobrin) [reply src/pages/documentation/building-mobile-extensions/interacting-with-rules.md:9]: there are few more details about how to create data elements, events etc for extensions, maybe we can preserve some of this in github / here

**Pull request merged**